RecoTracker/LSTCore/standalone/analysis/DNN/train_T4_DNN.ipynb

```diff
@@ -0,0 +1,1602 @@
+    "            1.0/outerRad,\n",
+    "            innerRad/outerRad,\n",
+    "            1.0/regRad,\n",
+    "            1.0/nonAnchorRegRad,\n",
+    "\n",
+    "            f1,\n",
+    "            p1,\n",
+    "            d1,\n",
+    "\n",
+    "            f2,\n",
+    "            p2,\n",
+    "            d2,\n",
+    "\n",
+    "            (f2 - f1),\n",
+    "            (p2 - p1),\n",
+    "            (d2 - d1),\n",
+    "        ]\n",
+    "\n",
+    "        # Use the abs eta value of first hit to select cut thresholds\n",
+    "        eta_iter.extend([np.abs(branches['t4_t3_0_eta'][event][idx0])])\n",
+    "        \n",
+    "        # Append the feature vector to the list\n",
+    "        features_list.append(features_iter)\n",
+    "        eta_list.append(eta_iter)\n",
+    "\n",
+    "# Convert the list of features to a NumPy array\n",
+    "features = np.array(features_list).T\n",
+    "eta_list = np.array(eta_list).T"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import torch\n",
+    "\n",
+    "# Stack features along a new axis to form a single array suitable for NN input\n",
+    "input_features_numpy = np.stack(features, axis=-1)\n",
+    "\n",
+    "# Identify rows with NaN or Inf values\n",
+    "mask = ~np.isnan(input_features_numpy) & ~np.isinf(input_features_numpy)\n",
+    "\n",
+    "# Apply mask across all columns: retain a row only if all its entries are neither NaN nor Inf\n",
+    "filtered_input_features_numpy = input_features_numpy[np.all(mask, axis=1)]\n",
+    "t4_isFake_filtered = (np.concatenate(branches['t4_pMatched']) <= 0.75)[np.all(mask, axis=1)]\n",
+    "t4_sim_vxy_filtered = np.concatenate(branches['t4_sim_vxy'])[np.all(mask, axis=1)]\n",
+    "\n",
+    "# Convert to PyTorch tensor when ready to use with NN\n",
+    "input_features_tensor = torch.tensor(filtered_input_features_numpy, dtype=torch.float32)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Using device: cuda\n",
+      "Total samples: 1946967\n",
+      "Fake count: 1932985\n",
+      "Real count: 13982\n",
+      "Prompt count: 2190\n",
+      "Displaced count: 11792\n",
+      "Initial dataset size: 1946967\n",
+      "Class distribution before downsampling - Fake: 1932985.0, Prompt: 2190.0, Displaced: 11792.0\n",
+      "Class distribution after downsampling - Fake: 966492.0, Prompt: 2190.0, Displaced: 11792.0\n"
+     ]
+    },
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Epoch [1/300], Train Loss: 0.8262, Test Loss: 0.7307\n",
+      "Epoch [2/300], Train Loss: 0.6984, Test Loss: 0.6723\n",
+      "Epoch [3/300], Train Loss: 0.6558, Test Loss: 0.6354\n",
+      "Epoch [4/300], Train Loss: 0.6422, Test Loss: 0.6504\n",
```

Comment:
is this needed in the release? check what we have in other training notebooks already in the release and compress
@GNiendorf please check and perhaps comment on the rest of the file